core/arch/arm/include/sm/optee_smc.h

```diff
@@ -559,6 +559,76 @@
 /* See OPTEE_SMC_CALL_WITH_REGD_ARG above */
 #define OPTEE_SMC_FUNCID_CALL_WITH_REGD_ARG	U(19)
 
+/*
+ * Secure watchdog management service
+ *
+ * Call requests usage:
+ * [in]		a0	SMC Function ID, OPTEE_SMC_WATCHDOG
+ * [out]	a0	PSCI error code return
+ *
+ * Watchdog commands
+ * Command SMCWD_INIT : Watchdog initialization
+ * [in]		a1	Set to 0
+ * [out]	a1	The minimal timeout value in seconds supported
+ *		a2	The maximal timeout value in seconds supported
+ *
+ * Return codes:
+ * PSCI_RET_SUCCESS - Command success
```

Comment:
PSCI?  The SMCs here use OPTEE_SMC_RETURN_*

Reply:
Keeping PSCI return code to be compatible with the existing service. Linux driver (and U-boot too) are using PSCI code return.

Reply:
That's a mess, to be honest. I don't see what PSCI has to do with the OPTEE_SMC range of SMCs, PSCI uses a different range of SMCs. When looking at the implementation I see that you still use `OPTEE_SMC_RETURN_UNKNOWN_FUNCTION`.

Reply:
That is true but this is the ABI described in the [Linux kernel driver](https://elixir.bootlin.com/linux/v6.3/source/drivers/watchdog/arm_smc_wdt.c#L45) (and the [U-Boot one](https://elixir.bootlin.com/u-boot/v2023.07-rc1/source/drivers/watchdog/arm_smc_wdt.c#L45) by mimethism). Should these be fixed?

Reply:
That's relevant information. This doesn't have anything to do with the OPTEE_SMC interface at all, in fact, the ABI is already defined. In the kernel, it's accessed as a SiP service unless the SMC ID is overridden in device tree. I believe we should be better off with implementing it in the SiP service range here too, defined in a separate .h file and so on.

Reply:
As we have all means in OP-TEE to provide such service in a native way. I think it's a shame we can't define a generic OP-TEE service for watchdog management. The main objective of this P-R is to have an standard SMC func ID for this instead of requiring each platform to redefine its own func ID for the very exact same purpose.
This P-R also clarifies the service ABI that is not  explicit when one reads OP-TEE OS and Linux kernel implementations.

Reply:
There is already one defacto standard SMC id `0x82003D06` used if `arm,smc-id` isn't found in device tree. If the SiP SMC range isn't good enough and you must use the Trusted OS SMC range instead please use a sub-range that doesn't overlap with the range used by the OPTEE_SMC ABI range. I don't see any benefit in mixing the ABI with the OPTEE_SMC ABI.

OPTEE_SMC ABI uses decimal `50` as the sub-range for requests from normal world and decimal `62` for some signaling to OPTEED in TF-A.
Google is using decimal `63` for the `NSSMC_OPTEED_CALL_LOAD_IMAGE` so perhaps it's best to stay out of that sub-range entirely too.
Maybe decimal `60` is a suitable sub-range.